packages/jetbrains-plugin/src/test/kotlin/com/mongodb/jbplugin/fixtures/AssertionExtensions.kt

```diff
@@ -86,6 +88,53 @@ fun eventually(
     }
 }
 
+/**
+ * Waits until the block function finishes successfully up to 1 second (or the provided timeout).
+ *
+ * Example usages:
+ *
+ * ```kt
+ * eventually {
+ *    verify(mock).myFunction()
+ * }
+ * // with custom timeout
+ * eventually(timeout = Duration.ofSeconds(5)) {
+ *    verify(mock).myFunction()
+ * }
+ * ```
+ *
+ * @param timeout
+ * @param fn
+ * @param recovery
+ */
+private fun eventuallySuspendable(
+    timeout: Duration = Duration.ofSeconds(10),
+    coroutineScope: TestScope? = null,
+    recovery: () -> Unit = {},
+    fn: suspend (Int) -> Unit,
+) {
```

Comment:
We need this because using runBlocking on eventually blocks the entire test so it never finishes. By making this suspendable, the function can yield to other coroutines and works as expected.